Remove audit components in favor of industry standard tools (#205)
ember-a11y/ember-a11y-testing: README.md

}
```

#### Setting Options

You can provide options to axe-core for your tests using the `setRunOptions` API. Options can
be set a few ways:

Globally:

```javascript
// tests/test-helper.js
import { setRunOptions } from 'ember-a11y-testing';

setRunOptions({
  rules: {
    region: { enabled: true },
  },
  checks: {
    'color-contrast': {
      options: {
        noScroll: true,
      },
    },
  },
});
```

Test module level:

```javascript
import { module, test } from 'qunit';
import { setRunOptions } from 'ember-a11y-testing';

module('some test module', function (hooks) {

>   hooks.before(function () {

Reviewer: Needs to be `beforeEach` (this is because we reset after each test):

```suggestion
  hooks.beforeEach(function () {
```

Author: Ya I originally was going to do this, but thought it made more sense to do it once rather than for each test invoked. I supposed it doesn't matter, but it's not clear to me what `getContext().owner` would return in `hooks.before`. What would it be?

    setRunOptions({
      rules: {
        region: { enabled: true },
      },
      checks: {
        'color-contrast': {
          options: {
            noScroll: true,
          },
        },
      },
    });
  });

  // ...
});
```

Individual test level:

```javascript
import { module, test } from 'qunit';
import { a11yAudit } from 'ember-a11y-testing/test-support';
import { setRunOptions } from 'ember-a11y-testing';

module('some test module', function (hooks) {
  test('some test', function (assert) {
    setRunOptions({
      rules: {
        region: { enabled: true },
      },
      checks: {
        'color-contrast': {
          options: {
            noScroll: true,
          },
        },
      },
    });

    // ...
    a11yAudit();
    // ...
  });

  // ...
});
```

> Options are automatically torn down when the test context's owner is destroyed.

Reviewer: ```suggestion
When using `setRunOptions` during a test, the options you set are automatically reset when the test completes.
```

### Development Usage

> While this addon previously included a number of components that would aid in identifying axe violations during development, those have been deprecated in favor of other, industry standard tools such as:

Reviewer: Let's reword this to be focused on the go forward users, not about migration to 4.0.0. We should author a custom transition guide for migrating from 3.x to 4.x and we can list the concrete things for those users in that guide.

Author: Sounds good.

  /* a11yCheck options */
}
```

##### Turning the audit off

Lastly, if you really find the audits to be cramping development, you can turn
them off via a simple boolean switch:

```javascript
turnAuditOff: true;
```

#### Environment Options

Each of the fine-grained component hooks above can instead be defined for ALL components inside of your application's `config/environment.js` file. Simply supply them in a `componentOptions` hash on the `ember-a11y-testing` property of `ENV`.

```javascript
ENV['ember-a11y-testing'] = {
  componentOptions: {
    turnAuditOff: false, // Change to true to disable the audit in development
    axeCallback: defaultAxeCallback,
    axeOptions: defaultAxeOptions,
    visualNoiseLevel: 2,
    axeViolationClassNames: ['alert-box', 'alert-box--a11y'],
  },
};
```

By example, to enable a specific rule and to set options for a specific check, you can:

```javascript
ENV['ember-a11y-testing']: {
  componentOptions: {
    axeOptions: {
      rules: {
        'region': { enabled: true } // enable region rule
      },
      checks: {
        'color-contrast': { options: { noScroll: true } } // disable scrolling of color-contrast check
      }

>     }

Reviewer 2: We should provide instructions on how to do this sort of thing in `test-helper.js` (assuming that's where it would go).

Reviewer: Ya, it would basically be:

```js
// tests/test-helper.js

// ...snip...
import { setRunOptions } from 'ember-a11y-testing';

setRunOptions({
  rules: {
    region: { enabled: true },
  },
  checks: {
    'color-contrast': {
      options: {
        noScroll: true,
      },
    },
  },
});
```

Author: Ah yes. Agree. This was weird as there was an odd interleaving of the documentation for the test-based code and the component-based code. I would have chosen to keep them very separate, but oh well.

I can update the README with this example that @rwjblue showed.

> - [axe Chrome extension](https://www.deque.com/axe/browser-extensions/) - a free axe browser extension ideal for development teams to test web applications to help identify and resolve common accessibility issues.

Reviewer 2: ```suggestion
- [**axe Chrome extension**](https://www.deque.com/axe/browser-extensions/) - a free axe browser extension ideal for development teams to test web applications to help identify and resolve common accessibility issues.
```

> 

Reviewer 2: We should also reference Microsoft's Accessibility Insights extension, which is the one we primarily use. I'd probably list it first.
```suggestion
[**Accessibility Insights for Web**](https://accessibilityinsights.io/docs/en/web/overview) - Accessibility Insights for Web helps developers find and fix accessibility issues in web apps and sites. This browser extension for Chrome and the new Microsoft Edge runs on Windows, MacOS, and Linux computers.

```

Reviewer: Ya, that's great to call out.

Author: Agree. Thanks for the callout, @drewlee!
